sa/saro.go

```diff
@@ -724,12 +724,12 @@ func (ssa *SQLStorageAuthorityRO) CountInvalidAuthorizations2(ctx context.Contex
 		status = :status AND
 		expires > :expiresEarliest AND
 		expires <= :expiresLatest AND
-		identifierType = :dnsType AND
-		identifierValue = :ident`,
+		identifierType = :identType AND
+		identifierValue = :identValue`,
 		map[string]interface{}{
 			"regID":           req.RegistrationID,
-			"dnsType":         identifierTypeToUint[string(identifier.TypeDNS)],
-			"ident":           ident.Value,
+			"identType":       identifierTypeToUint[ident.ToProto().Type],
```

Comment:
This will panic if `Type` isn't in the list. While we do validate `Type` earlier in the call stack, it's still good to do some extra validation at RPC boundaries when reasonable (e.g. core.IsAnyNilOrZero). In this case, I think we should do the lookup in this map earlier, using the `, ok` syntax, and return an error if the type isn't in the map.

Reply:
Great catch, thanks!